Framework/Core/test/test_TreeToTable.cxx

```diff
@@ -96,15 +96,17 @@ BOOST_AUTO_TEST_CASE(TreeToTableConversion)
   BOOST_REQUIRE_EQUAL(table->Validate().ok(), true);
   BOOST_REQUIRE_EQUAL(table->num_rows(), ndp);
   BOOST_REQUIRE_EQUAL(table->num_columns(), ncols);
-  BOOST_REQUIRE_EQUAL(table->column(0)->type()->id(), arrow::boolean()->id());
-  BOOST_REQUIRE_EQUAL(table->column(1)->type()->id(), arrow::float32()->id());
-  BOOST_REQUIRE_EQUAL(table->column(2)->type()->id(), arrow::float32()->id());
-  BOOST_REQUIRE_EQUAL(table->column(3)->type()->id(), arrow::float32()->id());
-  BOOST_REQUIRE_EQUAL(table->column(4)->type()->id(), arrow::float64()->id());
-  BOOST_REQUIRE_EQUAL(table->column(5)->type()->id(), arrow::int32()->id());
-  BOOST_REQUIRE_EQUAL(table->column(6)->type()->id(), arrow::fixed_size_list(arrow::float64(), nelem)->id());
-  BOOST_REQUIRE_EQUAL(table->column(7)->type()->id(), arrow::fixed_size_list(arrow::float32(), 96)->id());
-  BOOST_REQUIRE_EQUAL(table->column(8)->type()->id(), arrow::fixed_size_list(arrow::boolean(), 5)->id());
+
+  BOOST_REQUIRE(table->column(0)->type()->Equals(arrow::boolean()));
+  BOOST_REQUIRE(table->column(1)->type()->Equals(arrow::float32()));
+  BOOST_REQUIRE(table->column(2)->type()->Equals(arrow::float32()));
+  BOOST_REQUIRE(table->column(3)->type()->Equals(arrow::float32()));
+  BOOST_REQUIRE(table->column(4)->type()->Equals(arrow::float64()));
+  BOOST_REQUIRE(table->column(5)->type()->Equals(arrow::int32()));
+  BOOST_REQUIRE(table->column(6)->type()->Equals(arrow::fixed_size_list(arrow::float64(), nelem)));
+  BOOST_REQUIRE(table->column(7)->type()->Equals(arrow::fixed_size_list(arrow::boolean(), 5)));
+  BOOST_REQUIRE(table->column(8)->type()->Equals(arrow::fixed_size_list(arrow::float32(), 96)));
+  BOOST_REQUIRE(table->column(9)->type()->Equals(arrow::uint8()));
```

Comment:
The old version has the advantage that if something breaks we see both sides of the equation. Please do not change it.

Reply:
The old version compares only type-id, but not the actual type object. I can just put in both versions then, redundancy in tests is not a problem.